### ISSUE TITLE
AssertionStats warning on clang 8

### ISSUE THREAD

**Issue by jimon** [catchorg/Catch2#1655]
**Describe the bug**
When compiling with clang 8 following warnings are generated:

```
../include/reporters/../internal/catch_interfaces_reporter.h:83:25: warning: explicitly defaulted copy assignment operator is implicitly deleted [-Wdefaulted-function-deleted]
        AssertionStats& operator = ( AssertionStats const& ) = default;
                        ^
../include/reporters/../internal/catch_interfaces_reporter.h:87:25: note: copy assignment operator of 'AssertionStats' is implicitly deleted because field 'assertionResult' has a deleted copy assignment operator
        AssertionResult assertionResult;
                        ^
../include/reporters/../internal/catch_assertionresult.h:54:29: note: copy assignment operator of 'AssertionResult' is implicitly deleted because field 'm_resultData' has a deleted copy assignment operator
        AssertionResultData m_resultData;
                            ^
../include/reporters/../internal/catch_assertionresult.h:28:24: note: copy assignment operator of 'AssertionResultData' is implicitly deleted because field 'lazyExpression' has a deleted copy assignment operator
        LazyExpression lazyExpression;
                       ^
../include/reporters/../internal/catch_assertionhandler.h:32:25: note: 'operator=' has been explicitly marked deleted here
        LazyExpression& operator = ( LazyExpression const& ) = delete;
```

**Expected behavior**
No warnings.

**Reproduction steps**
Compile anything with clang 8 and Catch2, for example in Android Studio or Ubuntu.

**Platform information:**
 - Catch version: **v2.4.2**

**Comment by horenmar** [comment]
Can you try with current version?

**Comment by nelhage** [comment]
I can confirm this is resolved for me on catch2 `master` (01ef7076f50f5f2b481ddf082e1afca3c926983f)
